benchmarks/imagenet/vitb16/aim.py

```diff
@@ -0,0 +1,161 @@
+from typing import List, Tuple, Union
+
+import torch
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.nn import MSELoss
+from torch.optim import AdamW
+from torch.optim.optimizer import Optimizer
+
+from lightly.models import utils
+from lightly.models.modules import AIMPredictionHead, MaskedCausalVisionTransformer
+from lightly.models.utils import get_2d_sincos_pos_embed, random_prefix_mask
+from lightly.transforms import AIMTransform
+from lightly.utils.benchmarking import OnlineLinearClassifier
+from lightly.utils.scheduler import CosineWarmupScheduler
+
+
+class AIM(LightningModule):
+    def __init__(self, batch_size_per_device: int, num_classes: int) -> None:
+        super().__init__()
+        self.save_hyperparameters()
+        self.batch_size_per_device = batch_size_per_device
+
+        img_size = 224
+        self.patch_size = 14
+        self.num_patches = (img_size // self.patch_size) ** 2
+
+        vit = MaskedCausalVisionTransformer(
+            img_size=img_size,
+            patch_size=self.patch_size,
+            num_classes=num_classes,
+            embed_dim=1536,
+            depth=24,
+            num_heads=12,
+            qk_norm=False,
+            class_token=False,
+            no_embed_class=True,
+        )
+        # Use absolute positional embedding.
+        pos_embed = get_2d_sincos_pos_embed(
+            embed_dim=vit.embed_dim,
+            grid_size=int(self.num_patches**0.5),
+            cls_token=False,
+        )
+        vit.pos_embed.requires_grad = False
+        vit.pos_embed.data.copy_(torch.from_numpy(pos_embed).float().unsqueeze(0))
+
+        self.backbone = vit
+        self.projection_head = AIMPredictionHead(
+            input_dim=vit.embed_dim, output_dim=3 * self.patch_size**2
+        )
+
+        self.criterion = MSELoss()
+
+        self.online_classifier = OnlineLinearClassifier(
+            feature_dim=vit.embed_dim, num_classes=num_classes
+        )
+
+    def forward(self, x: Tensor) -> Tensor:
+        features = self.backbone.forward_features(x)
```

Comment:
This is equivalent to `mask=None`,right? Shouldn't we have `mask` as an argument here?

Reply:
Yes it is equivalent. The `forward` method in all the benchmarking modules is used to get the features for evaluation, in that case we want bidirectional attention as described in the paper.

Reply:
Updated :)